CMSIS-pack update for STM32WB-family

JanneKiiskila opened the pull request:
Update of the tools/arm_pack_manager/index.json -file for the
STM32WB-chipset family.

    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)


    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
   
### Reviewers 

@ARMmbed/team-st-mcd  @jeromecoutant  @adustm @adbridge  @0xc0170

Commit:
CMSIS-pack update for STM32WB-family

Update of the tools/arm_pack_manager/index.json -file for the
STM32WB-chipset family.

jeromecoutant commented on tools/arm_pack_manager/index.json:
```diff
@@ -462354,6 +462354,564 @@
+              "non_secure_callable": false
+            },
+            "start": 134217728,
+            "size": 16777216,
```
There is an issue (already reported to CMSIS ST team) with ROM size value,
Is it possible to set 1M instead of 16M ? (1048576 instead of 16777216)

JanneKiiskila replied:
Fixed now.

jeromecoutant commented on tools/arm_pack_manager/index.json:
```diff
@@ -462354,6 +462354,564 @@
+              "non_secure_callable": false
+            },
+            "start": 134217728,
+            "size": 16777216,
```
There is an issue (already reported to CMSIS ST team) with ROM size value,
Is it possible to set 1M instead of 16M ? (1048576 instead of 16777216)

JanneKiiskila replied:
It's likely wrong in the original CMSIS-pack itself then?
It should be fixed there, too.

jeromecoutant replied:
Yes, already reported a long time ago.... :-(
Don't understand yet if it is Keil or ST side...

0xc0170 replied:
@jeromecoutant Send me an email with details. I would assume the pack belongs to ST but if that is not the case, lets find out and have a fix there. I can  talk to CMSIS team

JanneKiiskila replied:
This page at least does also advertise it as 16 MB? 
https://www.keil.com/dd2/stmicroelectronics/stm32wb55rgvx/
If I'm hitting the right line... 

```
      "STM32WB55RGVx": {
        "name": "STM32WB55RGVx",
        "memories": {
          "IROM1": {
            "access": {
              "read": true,
              "write": false,
              "execute": true,
              "peripheral": false,
              "secure": false,
              "non_secure": false,
              "non_secure_callable": false
            },
            "start": 134217728,
            "size": 16777216,
            "startup": true,
            "default": true
```
Seems to be for some reason line 466474 in my editor.

JanneKiiskila replied:
However, in this page; https://www.digikey.co.uk/product-detail/en/stmicroelectronics/STM32WB55RGV6/497-18488-ND/9919553?utm_campaign=buynow&utm_medium=aggregator&WT.z_cid=ref_findchips_standard&utm_source=findchips - it is indeed listed as 1 MB part, so I think the source information going to Keil has to be wrong. But, we can fix it here.

JanneKiiskila replied:
Fixed now.

Commit:
Fix ROM size for STM32WB55RGVx (16 MB -> 1 MB)

Per feedback from STM the correct ROM size is 1 MB, instead of
16 MB. The KEIL source information is (in the pack itself) wrong, since
the KEIL webpage lists it as a 16 MB part, too - but if you look into
other sources - it is indeed 1 MB.

Commit:
Sector size definitions to index.json/STM32WB5xx

Adding manually the sector sizes now. The reference manual states
these chips have a fairly nice memory map as sectors are evenly sized
and fairly small. Quote from reference manual;

"The Flash memory is organized as follows:
• A main memory block containing 256 pages of 4 KB, each page with eight rows of
512 bytes."

https://www.st.com/content/ccc/resource/technical/document/reference_manual/group0/83/cf/94/7a/35/a9/43/58/DM00318631/files/DM00318631.pdf/jcr:content/translations/en.DM00318631.pdf
- chapter 3.3.1 Flash memory organization
- internal flash starts at address 0x0800 0000

Commit:
Size & sectors also for STM32WB55VGYx

The size was still 16 megabytes, while in reality it is one megabyte.
Added the sectors, too (see previous commit msg for details).

jeromecoutant reviewed (approved):
Quick parsing script:
- STM32WB55CCUx : core  Cortex-M4F rom   262144 (256 KB) (  0x40000) at 0x8000000 ram1  131072 (128 KB) ( 0x20000) at 0x20000000  sectors: (4 KB 0x8000000)
- STM32WB55CEUx : core  Cortex-M4F rom   524288 (512 KB) (  0x80000) at 0x8000000 ram1  262144 (256 KB) ( 0x40000) at 0x20000000  sectors: (4 KB 0x8000000)
- STM32WB55CGUx : core  Cortex-M4F rom  1048576 (1.0 MB) ( 0x100000) at 0x8000000 ram1  262144 (256 KB) ( 0x40000) at 0x20000000  sectors: (4 KB 0x8000000)
- STM32WB55RCVx : core  Cortex-M4F rom   262144 (256 KB) (  0x40000) at 0x8000000 ram1  131072 (128 KB) ( 0x20000) at 0x20000000  sectors: (4 KB 0x8000000)
- STM32WB55REVx : core  Cortex-M4F rom   524288 (512 KB) (  0x80000) at 0x8000000 ram1  262144 (256 KB) ( 0x40000) at 0x20000000  sectors: (4 KB 0x8000000)
- STM32WB55RGVx : core  Cortex-M4F rom  1048576 (1.0 MB) ( 0x100000) at 0x8000000 ram1  262144 (256 KB) ( 0x40000) at 0x20000000  sectors: (4 KB 0x8000000)
- STM32WB55VCYx : core  Cortex-M4F rom   262144 (256 KB) (  0x40000) at 0x8000000 ram1  131072 (128 KB) ( 0x20000) at 0x20000000  sectors: (4 KB 0x8000000)
- STM32WB55VEYx : core  Cortex-M4F rom   524288 (512 KB) (  0x80000) at 0x8000000 ram1  262144 (256 KB) ( 0x40000) at 0x20000000  sectors: (4 KB 0x8000000)
- STM32WB55VGYx : core  Cortex-M4F rom  1048576 (1.0 MB) ( 0x100000) at 0x8000000 ram1  262144 (256 KB) ( 0x40000) at 0x20000000  sectors: (4 KB 0x8000000)
